Arrowgene.Ddon.GameServer/Characters/JobManager.cs

```diff
@@ -522,34 +522,16 @@ public void RemoveAbility(IDatabase database, CharacterCommon character, byte sl
             // TODO: Performance
             lock (equippedAbilities)
             {
-                // Need to shift all the abilities over so they are in the new order reflecting in the UI.
-                // We keep track of 2 "pointers" one which is currently iterating over the whole list and
-                // another which is keep tracking of the current spot which can be inserted into.
-                // Also, we can make some assumptions to make this operation less wasteful.
-                // 1. Assume all elements before the one removed are already in order properly.
-                // 2. Due to point #1, we only need to shift elements starting from slotNo (in the array)
-                // 3. There are no "holes" in the list (cases where we have -- object, null, object).
-                // 4. Due to #4, once we find the first NULL element, we can stop
-                //
-
-                // Set the slot to null to handle the edgecase of removing the last element of the list
+                // Seems client wants us to keep the abilities where it was added
+                // for the first time. So just null out the ability we want to remove
+                // in the position in the list. When inserting into the DB, we need to
+                // be careful and skip over null entries and continue to increment the
+                // slot index.
                 equippedAbilities[slotNo - 1] = null;
```

Comment:
And in memory too. You have to shift the augs after the null one slot down (one index down the list)

Reply:
I'm going to pull this change into a separate PR so it doesn't get in the way of this one. Will need to do more investigation of the current implementation and behavior and eventually figure out the proper way to resolve this exception and get the desired behavior when removing the augment.